lightning/restore/restore.go

```diff
@@ -1296,6 +1299,15 @@ func (rc *RestoreController) checkRequirements(_ context.Context) error {
 	return rc.backend.CheckRequirements()
 }
 
+func (rc *RestoreController) setGlobalVariables(ctx context.Context) error {
+	// try to enable new collation
+	enabled := ObtainNewCollationEnabled(ctx, rc.tidbMgr.db)
+	if enabled {
+		collate.EnableNewCollations()
+	}
```

Comment:
can we *disable* collation when `!enabled`? we can't be stuck in new collation mode in server mode (serving multiple tasks).